libraries/AP_Compass/AP_Compass.cpp

```diff
@@ -541,6 +537,51 @@ const AP_Param::GroupInfo Compass::var_info[] = {
     // @User: Advanced
     AP_GROUPINFO("OPTIONS", 43, Compass, _options, 0),
 
+#if COMPASS_MAX_UNREG_DEV > 0
+    // @Param: DEV_ID4
+    // @DisplayName: Compass4 device id
+    // @Description: Extra 4th compass's device id.  Automatically detected, do not set manually
+    // @ReadOnly: True
+    // @User: Advanced
+    AP_GROUPINFO("DEV_ID4", 44, Compass, extra_dev_id[0], 0),
+#endif // COMPASS_MAX_UNREG_DEV
+
+#if COMPASS_MAX_UNREG_DEV > 1
+    // @Param: DEV_ID5
+    // @DisplayName: Compass5 device id
+    // @Description: Extra 5th compass's device id.  Automatically detected, do not set manually
+    // @ReadOnly: True
+    // @User: Advanced
+    AP_GROUPINFO("DEV_ID5", 45, Compass, extra_dev_id[1], 0),
+#endif // COMPASS_MAX_UNREG_DEV
+
+#if COMPASS_MAX_UNREG_DEV > 2
+    // @Param: DEV_ID6
+    // @DisplayName: Compass6 device id
+    // @Description: Extra 6th compass's device id.  Automatically detected, do not set manually
+    // @ReadOnly: True
+    // @User: Advanced
+    AP_GROUPINFO("DEV_ID6", 46, Compass, extra_dev_id[2], 0),
+#endif // COMPASS_MAX_UNREG_DEV
+
+#if COMPASS_MAX_UNREG_DEV > 3
```

Comment:
For the purpose of reducing parameters, I think we could limit the maximum number of compasses to 6.  So that's 3 that we can actually use and then another 3 that are there but unusable.  Surely 6 is enough isn't it?